### PR TITLE
More global variables - issue #388

### PR DESCRIPTION
**Pull request by jfrickmann**
I changed MAX_GVARS to 16. This caused some size checks and asserts to fail, so I changed those as well. I do not know what consequences this may have - does it corrupt model files or the EEPROM?
Therefore: DO NOT load this on your transmitter at this time!!

### CODE REVIEW

**Comment by wimalopaan** [comment]
I did that in the past as well for my local branch, and this invalidates eeprom and model bin files.

There was some discussion on discord, that this should be "safe" with yaml. safe in that respect non existent entries are silently ignored, but I did not test that with yaml. 

I "abused" GVs to make a communication from widgets to mixers, but now we have the SHMs for that. In general I think we should increase the number, maybe only for colorlcd.

**Comment by jfrickmann** [comment]
That’s what I feared would happen. Is it only the color radios using YAML?

**Comment by wimalopaan** [comment]
No, also the BW are using YAML (if enabled). But asked about the bin files ...

The exchange between YAML-file with different GV numbers should work (in theory: didn't test that)

**Comment by raphaelcoeffic** [comment]
> That’s what I feared would happen. Is it only the color radios using YAML?

YAML is used on all targets.

**Comment by jfrickmann** [comment]
Well maybe, I should just try it out on my spare X9D+ and see what happens 😎 
First, backup, then upgrade to "stock" YAML, then build FW with 16 GVs, and see what happens when I fire it up...

**Comment by gagarinlg** [comment]
Good luck

**Comment by elecpower** [comment]
and don't forget all the changes you need to make to Companion when you get this working on the radios... ;-)

File: radio/src/storage/yaml/yaml_datastructs_funcs.cpp [hunk 1]
```diff
@@ -114,7 +114,13 @@ static inline void check_yaml_funcs()
 {
   static_assert(offsetof(ModuleData, ppm) == 4,"");
   check_size<ModuleData, 29>();
+
+#if defined(PCBHORUS) && defined(PCBX10)
   static_assert(MAX_GVARS == 16,"");
+#else
+  static_assert(MAX_GVARS == 9,"");
+#endif
```
Comment (raphaelcoeffic) [inline radio/src/storage/yaml/yaml_datastructs_funcs.cpp:122]: The proper way here is to check which of the functions reused from the conversion uses the old constant, and copy/adapt the function here and remove the check.

**Comment by jfrickmann** [comment]
I am closing this one - maybe we can get back to it later!

**Pull request closed without merging**